app/js/controllers/main.js

```diff
@@ -129,8 +136,44 @@ export default class MainController extends Controller {
   // directly, you'll end up with more than one gesture detector listening
   // at a time.
   _waitToBeOpened() {
-    Gesture.detect(this.openGesture)
-      .then(() => this._lazyLoadModules())
+    Gesture.detect(this.openGesture).then(() => this.open());
+  }
+
+  _waitToBeClosed() {
+    Gesture.detect(this.closeGesture).then(() => this.close());
+  }
+
+  _checkOpenFromLauncher() {
+    var requestXHR = new XMLHttpRequest();
+    requestXHR.open('GET', 'http://localhost:3215/request', true);
+    requestXHR.onload = () => {
+      console.log('Customizer Launcher HTTP response', requestXHR.responseText);
+      if (requestXHR.responseText !== this.manifestURL) {
+        return;
+      }
+
+      console.log('Opened "' + this.manifestURL + '" from Customizer Launcher');
+      this.open();
+
+      var confirmXHR = new XMLHttpRequest();
+      confirmXHR.open('GET', 'http://localhost:3215/confirm?url=' + this.manifestURL, true);
+
+      console.log('Sending HTTP request confirmation to Customizer Launcher');
+      confirmXHR.send();
+    };
+
+    console.log('Sending HTTP request check to Customizer Launcher');
```

Comment:
This is a lot of logging. I think it would be better to guard this behind a ```DEBUG``` flag and ```debug()``` function.